app/src/main/java/com/example/testjavaapp/service/BroadcastService.java

```diff
@@ -0,0 +1,99 @@
+package com.example.testjavaapp.service;
+
+import android.annotation.TargetApi;
+import android.app.Service;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.os.Build;
+import android.os.IBinder;
+import android.util.Log;
+
+import androidx.localbroadcastmanager.content.LocalBroadcastManager;
+import com.example.testjavaapp.ServiceActivity;
+import com.example.testjavaapp.data.DataHolder;
+import com.example.testjavaapp.util.DownloadCallback;
+import com.example.testjavaapp.util.DownloadImageThread;
+import com.example.testjavaapp.util.DownloadImageThreadComplete;
+
+import java.util.ArrayList;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.Semaphore;
+
+public class BroadcastService extends Service implements DownloadCallback {
+    private String LOG_TAG = "RecyclerView";
+    private ArrayList<String> urls;
+    private ArrayList<Bitmap> bitmaps;
+
+    private CyclicBarrier barrier;
+    private Semaphore semaphore;
+    private CountDownLatch countDownLatch;
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        LOG_TAG = "RecyclerView";
+        Log.d(LOG_TAG, "In onCreate");
+        bitmaps = new ArrayList<>();
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        Log.d(LOG_TAG, "In onStartCommand");
+
+        setDownloadData(intent);
+
+        return START_REDELIVER_INTENT;
+    }
+
+    private void setDownloadData(Intent intent) {
+
+        int barrierNum = intent.getIntExtra("barrier_num",0);
+        int semaphoreNum = intent.getIntExtra("semaphore_num",0);
+        urls = intent.getStringArrayListExtra("url_list");
+
+        barrier = new CyclicBarrier(barrierNum);
+        semaphore = new Semaphore(semaphoreNum,true);
+        countDownLatch = new CountDownLatch(barrierNum);
+
+        new Thread(new DownloadImageThreadComplete(this,countDownLatch)).start();
```

Comment:
DownloadImageThreadComplete, first of all - this is runnable, not thread. Second - specification of type (Thread, Runnable) must be at the end of the name

Reply:
Fixed